xarray/backends/common.py

```diff
@@ -267,13 +268,23 @@ def robust_getitem(array, key, catch=Exception, max_retries=6, initial_delay=500
             time.sleep(1e-3 * next_delay)
 
 
-class BackendArray(NdimSizeLenMixin, indexing.ExplicitlyIndexed):
+class BackendArray(ABC, NdimSizeLenMixin, indexing.ExplicitlyIndexed):
```

Comment:
As `__getitem__` is required, I feel like `BackendArray` should always have been an ABC.

Reply:
This class is public API and this is a backwards incompatible change.

Reply:
It is technically, but only if someone is using this class in a way counter to what the docs explicitly tell you to do (i.e. subclass it).

Regardless this is orthogonal to the rest of the PR, I can remove it, I was just trying to clean up bad things I found.

Reply:
Reverted in https://github.com/pydata/xarray/pull/10327/commits/6c47e3f41bb4934959df6eb14acbb5dc0caf15d1